src/force_fields_input.F

```diff
@@ -2395,15 +2396,45 @@ SUBROUTINE read_nequip_data(nequip)
       READ (cutoff_str, *) nequip%rcutsq
       nequip%rcutsq = cp_unit_to_cp2k(nequip%rcutsq, nequip%unit_coords)
       nequip%rcutsq = nequip%rcutsq*nequip%rcutsq
-      nequip%unit_coords_val = 1.0_dp
       nequip%unit_coords_val = cp_unit_to_cp2k(nequip%unit_coords_val, nequip%unit_coords)
-      nequip%unit_forces_val = 1.0_dp
       nequip%unit_forces_val = cp_unit_to_cp2k(nequip%unit_forces_val, nequip%unit_forces)
-      nequip%unit_energy_val = 1.0_dp
       nequip%unit_energy_val = cp_unit_to_cp2k(nequip%unit_energy_val, nequip%unit_energy)
-      nequip%unit_cell_val = 1.0_dp
       nequip%unit_cell_val = cp_unit_to_cp2k(nequip%unit_cell_val, nequip%unit_cell)
 
+      ! look in config which contains all the .yaml file options to see if we use float32 or float64
+      config_str = torch_model_read_metadata(nequip%nequip_file_name, "config")
+      len_config = LEN_TRIM(config_str)
+      idx = INDEX(config_str, "default_dtype:")
```

Comment:
Parsing YAML like this is really brittle. Maybe you could ask the model makers to add all the information that you need as individual metadata fields?

Reply:
Thank you, I have put it into a subroutine instead of duplicating as suggested. There is ongoing development to make allegro work with lammps in double as well as single precision, so this might be done soon from their side. I will soon contact them about this also to let them know about the implementation. If parsing the YAML file is less than ideal, I could specify the `default_dtype` as a tag in the cp2k input file, as also done for the `ATOMS` tags where a list of chemical species is provided...